src/components/reusable/repeatInput/repeatInput.ts

```diff
@@ -0,0 +1,217 @@
+import { LitElement, html } from 'lit';
+import {
+  customElement,
+  property,
+  state,
+  query,
+  queryAssignedElements,
+} from 'lit/decorators.js';
+import Styles from './repeatInput.scss';
+import { classMap } from 'lit/directives/class-map.js';
+import { ifDefined } from 'lit/directives/if-defined.js';
+import clearIcon from '@carbon/icons/es/close/24';
+import '@kyndryl-design-system/shidoka-foundation/components/icon';
+import '../numberInput/numberInput';
+
+/**
+ * Text input.
+ * @fires on-input - Captures the input event and emits the selected value and original event details.
+ * @prop {string} pattern - RegEx pattern to validate.
+ * @prop {number} minLength - Minimum number of characters.
+ * @prop {number} maxLength - Maximum number of characters.
+ * @slot unnamed - Slot for label text.
+ * @slot icon - Slot for contextual icon.
+ */
+@customElement('kyn-repeat-input')
```

Comment:
This component might have been easier to build if you used kyn-text-input directly rather than copying all of it's internals. One thing we always try not to do is duplicate code: DRY (don't repeat yourself).